tests/component-tests/pager/js/PagedTableTests.js

```diff
@@ -218,17 +170,123 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+            return {text: data.animal};
+        });
+
+        /* Change the page size and verify whether the tooltips and links are synced accordingly */
+        fluid.each(sequencesOfPageSizes, function (pageSize) {
+            var expectedTooltips = fluid.tests.getTooltipContents(linksList, pageSize);
+            $.merge(sequence, [
+                {
+                    func: "fluid.changeElementValue",
+                    args: ["{pager}.dom.pageSize", pageSize]
+                },
+                {
+                    func: "fluid.tests.verifyPagerBar",
+                    args: ["{pager}", "top", pageSize]
+                },
+                {
+                    func: "fluid.tests.verifyPagerBar",
+                    args: ["{pager}", "bottom", pageSize]
+                },
+                {
+                    func: "fluid.tests.verifyToolTipsOfPagerBar",
+                    args: ["{pager}", "top", pageSize, expectedTooltips]
+                },
+                {
+                    func: "fluid.tests.verifyToolTipsOfPagerBar",
+                    args: ["{pager}", "bottom", pageSize, expectedTooltips]
+                }
+            ]);
+        });
+
+        return {
+            name: "Pager tooltip tests",
+            tests: [
+                {
+                    name: "Tooltip visibility and contents",
+                    sequence: sequence
+                }
+            ]
+        };
+    };
+
+    fluid.tests.getTooltipContents = function (listOfData, pageSize) {
+        var tooltipContents = [];
+        for (var i = 0; i < listOfData.length; i += pageSize) {
+            var tooltipContent = [];
+            var pageStartIndex = i;
+            var pageEndIndex = Math.min(pageStartIndex + pageSize - 1, listOfData.length - 1);
+            tooltipContent.push({nodeName: "b", nodeText: listOfData[pageStartIndex].text});
+            tooltipContent.push({nodeName: "b", nodeText: listOfData[pageEndIndex].text});
+            tooltipContents.push(tooltipContent);
+        }
+
+        return tooltipContents;
+    };
+
+    fluid.tests.getPagerBar = function (pager, location) {
+        return pager[location === "top" ? "pagerBar" : "pagerBar-1"];
+    };
+
+    fluid.tests.verifyPagerBar = function (pager, location, pageSize) {
+        var expectedNumberOfLinks = Math.ceil(pager.dataModel.length / pageSize);
+
+        //Verify the number of links in the pager bar
+        jqUnit.assertEquals("The number of links in the pager should be according to the page size",
+            expectedNumberOfLinks,
+            fluid.tests.getPagerBar(pager, location).locate("pageLinks").length);
+    };
+
+    fluid.tests.assertVisibleTips = function (pager, message, targetIds, expectedTooltip) {
+        fluid.tests.tooltip.assertVisible(message, pager, targetIds, null, function (tooltip) {
+            jqUnit.assertNode(tooltip.length, expectedTooltip, $("b", tooltip));
```

Comment:
What I'm seeing when I run these tests is something like the following:

> Expected number of nodes 1 - at sequence position 4 of 15
> 1: node 0: - node name expected value: b actual: b - at sequence position 4 of 15
> 1: node 0: - attribute nodeText expected value: dog actual: dog - at sequence position 4 of 15
> 1: node 1: - node name expected value: b actual: b - at sequence position 4 of 15
> 1: node 1: - attribute nodeText expected value: cat actual: cat - at sequence position 4 of 15

It's the "1" at the front that is particularly confusing. I don't know the I should equate this with a value from the preceding test. It's also confusing that if there are supposed to be only 1 node, why are two nodes tested. A real message should be able to explain what's being tested here.